onadata/apps/viewer/models/parsed_instance.py

```diff
@@ -165,7 +178,7 @@ def _get_sort_fields(sort):
 
 def get_sql_with_params(xform, query=None, fields=None, sort=None, start=None,
                         end=None, start_index=None, limit=None, count=None):
-    records = _get_instances(xform, start, end)
+    records, query = _get_instances(xform, start, end, query)
```

Comment:
I would have thought this is already being handled by `get_where_clause` even though the current implementation will return a JSON field filter rather than an `Instance.version` field filter. The following changes may be required for this field to be used.

```patch
index 38a64120d..475286a13 100644
--- a/onadata/apps/viewer/parsed_instance_tools.py
+++ b/onadata/apps/viewer/parsed_instance_tools.py
@@ -7,7 +7,8 @@ from onadata.libs.utils.common_tags import MONGO_STRFTIME
 KNOWN_DATES = ['_submission_time']
 NONE_JSON_FIELDS = {
     '_submission_time': 'date_created',
-    '_id': 'id'
+    '_id': 'id',
+    '_version': 'version'
 }
 
 
@@ -55,8 +56,12 @@ def _parse_where(query, known_integers, or_where, or_params):
                 else:
                     where_params.extend((field_key, unicode(_v)))
         else:
-            where.append(u"json->>%s = %s")
-            where_params.extend((field_key, unicode(field_value)))
+            if field_key in NONE_JSON_FIELDS:
+                where.append("{} = %s".format(NONE_JSON_FIELDS[field_key]))
+                where_params.extend([unicode(field_value)])
+            else:
+                where.append(u"json->>%s = %s")
+                where_params.extend((field_key, unicode(field_value)))
 
     return where + or_where, where_params + or_params
```